### ISSUE TITLE
Templates should be able to use #, affect the parent scope

### ISSUE THREAD

**Issue by zoechi** [dart-lang/angular_analyzer_plugin#167]
```html
<template #itemTemplate let-item="item" let-i="index">
  <list-item [label]="item.name" (click)="navigateToMenu(item)"></list-item>
</template>

<mdl-card *ngIf="projectService.project != null" class="mdl-shadow--2dp">
  <span card-title>Select</span>

  <div card-content>
    <list-component class="menus-list" [data]="projectService?.project?.menus"
                    [template]="itemTemplate" tabIndex="0"
                    [(ngModel)]="projectService.selectedMenu"
                    (ngModelChange)="navigateToMenu($event)"></list-component>

    <div class="buttons">
      <add-item-button (click)="addMenu()">Create a new menu</add-item-button>
      <remove-item-button (click)="removeMenu()"
                          [disabled]="projectService.selectedMenu == null">
        Delete the selected menu
      </remove-item-button>
    </div>
  </div>

</mdl-card>

<router-outlet></router-outlet>
```

results in 

```
ERROR: Undefined name 'itemTemplate'. ([fhir_client] lib/designer/webui/menu_editor/menus_editor_component.html:10)
```

for 

```html
 [template]="itemTemplate"
```

**Comment by MichaelRFairhurst** [comment]
Hmmmm....this is a tricky one.

Usually, templates denote scopes. The issue is that the template is only in scope within the template.

Ah, I think I see where my confusion comes from: `let-item` makes a variable within the scope of `<template>`, while `#itemTemplate` makes a variable within the scope outside `<template>`? @matanlurey  can you confirm?

**Comment by MichaelRFairhurst** [comment]
Ping @matanlurey  again

**Comment by zoechi** [comment]
Friendly ping. 
This is still an issue.

**Comment by MichaelRFairhurst** [comment]
Thanks for filing #321, that clarified this to me.

So, in a <template> tag, `#foo` and `let foo` are different, and `#foo` should affect the parent scope while `let` affects the child scope.

**Comment by zoechi** [comment]
Exactly, I added some explanation in #321 that should make it more clear.

**Comment by MichaelRFairhurst** [comment]
Awesome, that will definitely help!

**Comment by mk13** [comment]
Taking a look at this, will have a fix soon.

**Comment by mk13** [comment]
Should be fixed with PR #348 . Reopen if problem persists.
